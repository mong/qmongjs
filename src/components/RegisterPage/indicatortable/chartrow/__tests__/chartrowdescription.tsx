import ChartRowDescription from "../chartrowdescription";
import { render } from "@testing-library/react";

test("ChartRowDescription where date will change with time zone", () => {
  let delivery_time = new Date("1999-12-31T23:59:59.999Z");
  const { container } = render(
    <ChartRowDescription
      description_text="Denne kvalitetsindikatoren er definert som andel pasienter med..."
      delivery_time={delivery_time}
    />
  );

  expect(container).toMatchSnapshot();
});

test("ChartRowDescription with date equals zero and Norwegian characters", () => {
  let delivery_time = new Date(0);
  const { container } = render(
    <ChartRowDescription
      description_text="Vi prøver ågså særnårske bokstaver"
      delivery_time={delivery_time}
    />
  );

  expect(container).toMatchSnapshot();
});

test("ChartRowDescription with non-default title and +02:00 time zone", () => {
  let delivery_time = new Date("1979-03-10T23:59:59.000+02:00");
  const { container } = render(
    <ChartRowDescription
      description_text="Denne kvalitetsindikatoren er definert som andel pasienter med..."
      delivery_time={delivery_time}
      description_title="Dette er en annen tittel"
    />
  );

  expect(container).toMatchSnapshot();
});
