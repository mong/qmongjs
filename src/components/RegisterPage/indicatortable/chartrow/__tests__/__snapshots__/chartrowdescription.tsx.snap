// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartRowDescription where date will change with time zone 1`] = `
<div>
  <div
    class="description_container"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
        >
          <div
            class="description_title"
          >
            Om kvalitetsindikatoren
          </div>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <p
                  style="margin: 0px;"
                >
                  Denne kvalitetsindikatoren er definert som andel pasienter med...
                </p>
                <p>
                  Data ble sist oppdatert
                   
                  1. januar 2000
                   
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ChartRowDescription with date equals zero and Norwegian characters 1`] = `
<div>
  <div
    class="description_container"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
        >
          <div
            class="description_title"
          >
            Om kvalitetsindikatoren
          </div>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <p
                  style="margin: 0px;"
                >
                  Vi prøver ågså særnårske bokstaver
                </p>
                <p>
                  Data ble sist oppdatert
                   
                  1. januar 1970
                   
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ChartRowDescription with non-default title and +02:00 time zone 1`] = `
<div>
  <div
    class="description_container"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
        >
          <div
            class="description_title"
          >
            Dette er en annen tittel
          </div>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <p
                  style="margin: 0px;"
                >
                  Denne kvalitetsindikatoren er definert som andel pasienter med...
                </p>
                <p>
                  Data ble sist oppdatert
                   
                  10. mars 1979
                   
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
